ros/src/twist_controller/twist_controller.py

```diff
@@ -2,13 +2,108 @@
 GAS_DENSITY = 2.858
 ONE_MPH = 0.44704
 
+from yaw_controller import YawController
+from pid import PID
+from lowpass import LowPassFilter
+import rospy
+
 
 class Controller(object):
-    def __init__(self, *args, **kwargs):
-        # TODO: Implement
-        pass
-
-    def control(self, *args, **kwargs):
-        # TODO: Change the arg, kwarg list to suit your needs
-        # Return throttle, brake, steer
-        return 1., 0., 0.
+    def __init__(self, vehicle_mass, fuel_capacity, brake_deadband, decel_limit,
+        accel_limit, wheel_radius, wheel_base, steer_ratio, max_lat_accel, max_steer_angle):
+
+        self.yaw_controller = YawController(wheel_base, steer_ratio, 0.1, max_lat_accel, max_steer_angle)
+
+        #ajaffer: Tried with Highway track only, for parking lot we might
+        # need to go to 0.2
+        # self.max_throttle = 0.2
+        self.max_throttle = 1.0
+
+        kp = 0.3
```

Comment:
How did you arrive at these values? Can we tweak them to reduce the start-stop behavior at low speeds?

Reply:
I started with the values from lecture notes, experimented with different set of values using the ROS bag reference implementation and plotted the graphs of brakes, throttles and steering, these were the best values I settled on. The car also drives good using these values, we can tweak them if needed. These are no listed in styx.launch file as params.